packages/loot-core/src/server/accounts/parse-file.js

```diff
@@ -57,6 +59,106 @@ async function parseCSV(filepath, options = {}) {
+  let out = [];
+
+  data.forEach(element => {
+    let tmp = {};
+    tmp.date = element.ValDt.Dt._text;
+    
+    if (element.CdtDbtInd._text == "DBIT") {
+      tmp.amount = -looselyParseAmount(element.Amt._text);
+    } else {
+      tmp.amount = looselyParseAmount(element.Amt._text);
+    }
+    
+    if (typeof element.NtryDtls === 'undefined' || element.NtryDtls === null) {
+      tmp.payee_name = null;
+      
+      if (typeof element.BkTxCd !== 'undefined' || element.BkTxCd !== null) {
+        
+        // Specific for ABNAMRO, retrieve Payee name when paying by card in a store
+        if (element.BkTxCd.Domn.Cd._text == 'PMNT' &&
+        element.BkTxCd.Domn.Fmly.Cd._text == 'CCRD' &&
+        element.BkTxCd.Domn.Fmly.SubFmlyCd._text == 'POSD' &&
+        element.AddtlNtryInf._text.toString().startsWith('BEA, Betaalpas')) {
+          var arr = element.AddtlNtryInf._text.toString().split(",");
+          tmp.payee_name = arr[1].toString().replace(' Betaalpas', "").trim();
+        } else {
+          tmp.payee_name = null;
+        }
+      }
+      
+      tmp.imported_payee = tmp.payee_name;
+      tmp.notes = element.AddtlNtryInf._text;
+      
+    } else {
+      if (element.CdtDbtInd._text == "DBIT") {
+        tmp.payee_name = element.NtryDtls.TxDtls.RltdPties.Cdtr.Nm._text;
+      } else {
+        tmp.payee_name = element.NtryDtls.TxDtls.RltdPties.Dbtr.Nm._text;
+      }
+      tmp.imported_payee = tmp.payee_name;
+      
+      if (typeof element.NtryDtls.TxDtls.RmtInf === 'undefined' || element.NtryDtls.TxDtls.RmtInf === null) {
+        tmp.notes = (typeof element.AddtlNtryInf === 'undefined' || element.AddtlNtryInf === null) ? null : element.AddtlNtryInf._text;
+      } else {
+        tmp.notes = element.NtryDtls.TxDtls.RmtInf.Ustrd._text;
+      }
+    }
+    
+    // AcctSvcrRef seems to be unreliable as a unique ID for entries, create our own
+    // tmp.imported_id = element.AcctSvcrRef.toString();
+    tmp.imported_id = `${tmp.date}_${tmp.payee_name}_${tmp.amount}`;
+    console.log("DEBUG: %s", tmp.imported_id);
+    
+    out.push(tmp);
+  }
+  
+  );
+  
+  return out;
+}
+
+async function parseCAMT053(filepath, options = {}) {
+  let errors = [];
+  let contents = await fs.readFile(filepath);
+
+  let data;
+  try {
+    var convert = require('xml-js');
+
+    data = convert.xml2js(contents, {compact: true, spaces: 4, ignoreComment: true, trim: true});
+    if (Array.isArray(data.Document.BkToCstmrStmt.Stmt.Ntry)) {
+      data = data.Document.BkToCstmrStmt.Stmt.Ntry;
+    } else {
+      data = [data.Document.BkToCstmrStmt.Stmt.Ntry];
+    }
+
+    // CAMT.053 entries have quite a bit of variety so we'll process them seperately
+    data = parseCAMT053Entries(data);
+
+  } catch (err) {
+    errors.push({
+      message: 'Failed parsing: ' + err.message,
+      internal: err.message
+    });
+    return { errors, transactions: [] };
+  }
+
+  return {
+    errors,
+    transactions: data.map(trans => ({
```

Comment:
It seems like this `map` can be removed? It doesn’t look like it’s doing anything.